compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -928,9 +934,72 @@ trait Implicits:
           if normalImports.isEmpty then suggestedImports else normalImports
         end hiddenImplicitsAddendum
 
-        msg(userDefined.getOrElse(
-          em"no implicit argument of type $pt was found${location("for")}"))() ++
-        hiddenImplicitsAddendum
+
+        /**
+         * Extracting the message from a method parameter, e.g. in
+         *
+         * trait Foo
+         * 
+         * def foo(implicit @annotation.implicitNotFound foo: Foo)
+         */
+        def userDefinedParamMessage = paramSymWithMethodTree.flatMap{ (sym, tree) =>
+          userDefinedMsg(sym, defn.ImplicitNotFoundAnnot).map{ rawMsg =>
```

Comment:
Space before braces (same in a few other places):
```suggestion
        def userDefinedParamMessage = paramSymWithMethodTree.flatMap { (sym, tree) =>
          userDefinedMsg(sym, defn.ImplicitNotFoundAnnot).map { rawMsg =>
```